CONTRIBUTING.rst

```diff
@@ -177,6 +177,30 @@ Build the docs via:
 
    $ nox -s docs
 
+*************************
+Samples and code snippets
+*************************
+
+Code samples and snippets live in the `samples/` catalogue. Feel free to
+provide more examples, but make sure to write tests for those examples.
+Each folder containing example code requires its own `noxfile.py` script
+which automates testing. If you decide to create a new folder, you can
+base it on the `samples/snippets` folder (providing `noxfile.py` and
+the requirements files).
+
+The tests will run against a real Google Cloud Project, so you should
+configure them just like the System Tests.
+
+- To run sample tests, you can execute::
```

Comment:
This seems explicit, but this is a single sort of sample. There are many in bigtable. When I read this it sort of reads like the only samples are samples/snippets.

Reply:
cc: @busunkim96 @parthea is this unique to bigtable (having so many other directories in samples)

Reply:
- `python-spanner` has one directory of samples, oddly enough nested under [`samples/samples`]().  Logically they could be broken up, rooted at just `samples/`, but that would break docs.
- `python-bigquery` has an odd mix:  lots of samples in just `samples/`, but the test drivers are in `samples/tests/`, and the snippets are `samples/snippets/`, along with their test drivers.
- `python-pubsub` only has runnable samples in `samples/snippets/`, including their test drivers.
- `python-storage` has no samples.

The generated libraries have, interestingly, almost as wide a range.

Reply:
In BigQuery, the goal is to have samples that just require the `google-cloud-bigquery` library in `samples/snippets`, but with other directories like `samples/geography` and `sample/pandas` for samples that have extra dependencies. 

Just filed https://github.com/googleapis/python-bigquery/issues/790 to actually document this goal.